pkg/utils/utils.go

```diff
@@ -284,6 +284,39 @@ func CleanCachedNetConf(cRefPath string) error {
 	return nil
 }
 
+// SetVFEffectiveMAC will try to set the mac address on a specific VF interface
+// the function will also validate that the mac address was configured as expect
+// it will return an error if it didn't manage to configure the vf mac address
+// or the mac is not equal to the expect one
+// retries 20 times and wait 100 milliseconds
+func SetVFEffectiveMAC(netLinkManager NetlinkManager, netDeviceName string, macAddress string) error {
+	hwaddr, err := net.ParseMAC(macAddress)
+	if err != nil {
+		return fmt.Errorf("failed to parse MAC address %s: %v", macAddress, err)
+	}
+
+	linkObj, err := netLinkManager.LinkByName(netDeviceName)
+	if err != nil {
+		return err
+	}
+
+	return Retry(20, 100*time.Millisecond, func() error {
+		if err := netLinkManager.LinkSetHardwareAddr(linkObj, hwaddr); err != nil {
+			return err
+		}
+
+		linkObj, err := netLinkManager.LinkByName(netDeviceName)
```

Comment:
do we really need to get linkObj here ? instead of reusing from L298 ?

and if yes, i think its cleaner to:

1. remove LinkByName from L298
2. call LinkByName first in this func.

WDYT?

or is there some case im missing ?

Reply:
the idea is to verify that the above func (LinkSetHardwareAddr) changed the mac. It needs to fetch again for that. If you add it first, it will do more work. Fetch, modify and then fetch again to verify.

Reply:
ah ! yes.  momentary loss of knowledge  on how this works lol :)

Reply:
please check now I prefer to use orgLinkObj for the attr and have a new one to recheck the mac address everytime